viewer/Viewer.h

```diff
@@ -111,23 +124,58 @@ namespace Enki
 		typedef QMapIterator<const std::type_info*, const std::type_info*> ManagedObjectsAliasesMapIterator;
 		ManagedObjectsAliasesMap managedObjectsAliases;
 		
+		typedef std::pair<QString, unsigned int> viewerMessage;
+		std::list<viewerMessage> messageList;
+		QString controlError1;	// dislpayed if user try to move an object in trackball mode : action forbiden to prevent glitchies
+		QString controlError2;	// dislpayed if user try to move camera in trackball mode
+		QString controlHelp;	// dislpayed if user press F1
+
+		struct ExtendedAttributes
+		{
+			bool movableByPicking;
+
+			ExtendedAttributes():movableByPicking(false){};
+		};
+		std::map<PhysicalObject*, ExtendedAttributes> objectExtendedAttributesList;
```

Comment:
for the UserData we can't use this technic because every object drawable need this attribute, which is too bad in term of complexity with a map or set container.